tests/integration/data/kfp_1.7_stable_generic_install.yaml.j2

```diff
@@ -0,0 +1,198 @@
+bundle: kubernetes
+applications:
+  argo-controller:
+    charm: argo-controller
+    channel: 3.3/stable
+    revision: 236
+    series: focal
+    resources:
+      oci-image: 185
+    scale: 1
+    constraints: arch=amd64
+  argo-server:
+    charm: argo-server
+    channel: 3.3/stable
+    revision: 185
+    series: focal
+    resources:
+      oci-image: 186
+    scale: 1
+    constraints: arch=amd64
+  dex-auth:
+    charm: dex-auth
+    channel: 2.31/stable
+    revision: 224
+    series: focal
+    resources:
+      oci-image: 318
+    scale: 1
+    options:
+      public-url: http://10.64.140.43.nip.io
+      static-password: admin
+      static-username: admin
```

Comment:
We want this to configure Dex's static username and password, the ones used in the test are for authenticating (so equivalent of you introducing your passwd/username). I could read the values in this bundle definition and use them in the test file, if that sounds better.

Reply:
A note here that we could avoid Dex/AuthService in testing altogether by:
1. Port-forwarding the KFP K8s Service to localhost
2. Updating the client to use that host, but manually setting the userid header (if needed)

We had done this in upstream KF tests
https://github.com/kubeflow/manifests/blob/master/tests/e2e/hack/proxy_pipelines.sh#L4
https://github.com/kubeflow/manifests/blob/master/tests/e2e/mnist.py#L48-L51

But if you've tested and are more comfortable with the current implementation we can do the above in a follow-up improvement PR later on

Reply:
(bringing conversation over from [this thread](https://github.com/canonical/kfp-operators/pull/238#discussion_r1252298756) too, which is nearly the same)

Thinking about what functionality the dependencies bring:
* istio pilot+gateway: 
  * the external gateway url (so we don't need a port-forward)
  * the additional `kubeflow-userid` header **iff** we also have dex+auth
* dex+auth:
  * the requirement that users authenticate
  * the `kubeflow-userid` header

From that, I think we have a few options:
1. use dex+auth+istio
  * we access through the external url and log in via auth (like this pr already had)
  * `kubeflow-userid` is assigned automatically for us
2. omit dex/auth, but keep istio
  * we do not get `kubeflow-userid` automatically, so we need to add the header ourselves
  * we can connect to the external url (but this might remove the `kubeflow-userid` header)
3. omit dex/auth/istio
  * we can connect to kfp directly via port-forwarding
  * we must add the kubeflow-userid-header

I don't know which is better really.  (1) is closest to charmed kubeflow, feels like it adds more dependencies than the value we get from them.  (2) means we dont have to port-forward, but otherwise doesn't add much.  (3) makes connecting harder.  They're all bad in their own way :D

Reply:
But maybe I'm over complicating things, at least for this PR.  It feels heavy to deploy istio+dex+auth just so that we also need to log in, but then otherwise use kfp without caring about those tools.  But on the other side, it is simpler to implement because we're used to it.  So I trust your judgement in picking which feels best

Reply:
This conversation does not apply anymore to the current testing setup. Feel free to re-open if further discussion is needed.